### PR TITLE
Usersearch: Convert to TSX

### PR DESCRIPTION
**Pull request by KrisXV**

### CODE REVIEW

**Comment by AnnikaCodes** [comment]
This seems okay to me at a glance, but I haven't learned Preact yet so I'm not sure I feel comfortable merging this unilaterally.

File: server/chat-plugins/usersearch.tsx [hunk 1]
```diff
@@ -7,76 +7,101 @@ export const nameList = new Set<string>(JSON.parse(
...
-			results.online.push(`${!page ? ONLINE_SYMBOL : ''} ${curUser.name}`);
-		} else {
-			results.offline.push(`${!page ? OFFLINE_SYMBOL : ''} ${curUser.name}`);
+class PunishmentHTML extends Chat.JSX.Component<{userid: ID, target: string}> {
+	render() {
+		const {userid, target} = {...this.props};
+		const buf = [];
+		for (const cmdName of ['Forcerename', 'Namelock', 'Weeknamelock']) {
+			// We have to use dangerouslySetInnerHTML here because otherwise the `value`
+			// property of the button tag is auto escaped, making &#10; into &amp;#10;
```
Comment (Zarel) [inline server/chat-plugins/usersearch.tsx:16]: Why not just unescape it? Use `\n` instead of `&#10;`? Would that work?

Comment (KrisXV) [reply server/chat-plugins/usersearch.tsx:16]: I tried that, it doesn't work.

**Comment by AnnikaCodes** [comment]
Thanks!

**Pull request merged**